core/models.py

```diff
@@ -182,6 +184,96 @@ class Child(models.Model):
     picture = models.ImageField(
         blank=True, null=True, upload_to="child/picture/", verbose_name=_("Picture")
     )
+    wake_window = models.DurationField(
+        blank=True, null=True, verbose_name=_("Wake Window")
+    )
+
+    class NumberOfNaps(enum.Enum):
+        AUTO = 1
+        ZERO = 0
+
+        ONE = 100
+        ONE_OR_TWO = 150
+        TWO = 200
+        TWO_OR_THREE = 250
+        THREE = 300
+        THREE_OR_FOUR = 350
+        FOUR = 400
+
+        __labels__ = {
+            AUTO: _("Auto"),
+            ZERO: _("0"),
+            ONE: _("One"),
+            ONE_OR_TWO: _("1 or 2"),
+            TWO: _("2"),
+            TWO_OR_THREE: _("2 or 3"),
+            THREE: _("3"),
+            THREE_OR_FOUR: _("3 or 4"),
+            FOUR: _("4"),
+        }
+
+        @classmethod
+        def auto(cls, child, now=None):
+            """
+            0 to 3 months	3 to 4 (0-90 days)
+            4 to 7 months	2 to 3 (91-210 days)
+            8 to 12 months	2 (211-365 days)
+            12 months to 18 months	1-2 (366-545 days)
+            18 months to 3 years	1 (546-1095 days)
+            """
+            now = now or timezone.localtime()
+            age_in_days = (now.date() - child.birth_date).days
+            if age_in_days < 90:
+                return cls.THREE_OR_FOUR
+            if age_in_days < 210:
+                return cls.TWO_OR_THREE
+            if age_in_days < 365:
+                return cls.TWO
+            if age_in_days < 545:
+                return cls.ONE_OR_TWO
+            if age_in_days < 1095:
+                return cls.ONE
+            return cls.ZERO

```

Comment:
What is the basis for this data? I recall there there being a pretty decent standard for this back when I had younger kids. It would be good to document here what our sources are.

Reply:
Honestly a combination of some article that I googled, github copilot, and chatgpt might have even had a say in it. I can do some digging and come up with a more reliable source, or at least link to a reasonable article.

Reply:
Haha yeah let’s try to find a good source to rely on. I’ll do some looking as well. I try to use clearly well researched data from sources like government bodies or NGOs when possible.

Reply:
Ok so here is a WHO source _and_ the important thing I know I was forgetting... the _number_ of naps per day doesn't really matter. The important thing is just the total amount of sleep time: https://www.who.int/tools/your-life-your-health/life-phase/newborns-and-children-under-5-years/making-sure-newborns-and-children-under-5-years-sleep-safely (and more deeply: https://iris.who.int/bitstream/handle/10665/325147/WHO-NMH-PND-2019.4-eng.pdf).

So the WHO says:

- 0 - 3 months: 14 – 17 hours
- 4 - 11 months: 12 – 16 hours
- 1 - 2 years: 11 – 14 hours
- 3+ years: 10–13 hours

So that's where this predictive stuff gets a little tricky... I don't really want the app to prescribe something that isn't backed by strong research and guidance (e.g., number of naps per day as opposed to number of hours of sleep per day).

Can we think of a way to re-imagine this feature with a focus on recommended hours of sleep per day?